pkg/eval/json_evaluator_model.go

```diff
@@ -8,45 +8,139 @@ import (
 	"github.com/open-feature/flagd/pkg/logger"
 )
 
-type Flags struct {
-	Flags map[string]Flag `json:"flags"`
+type Flag struct {
+	State          string          `json:"state"`
+	DefaultVariant string          `json:"defaultVariant"`
+	Variants       map[string]any  `json:"variants"`
+	Targeting      json.RawMessage `json:"targeting,omitempty"`
+	Source         string          `json:"source"`
 }
 
 type Evaluators struct {
 	Evaluators map[string]json.RawMessage `json:"$evaluators"`
 }
 
-func (f Flags) Merge(logger *logger.Logger, source string, ff Flags) (Flags, map[string]interface{}) {
+type Flags struct {
+	Flags map[string]Flag `json:"flags"`
+}
+
+// Add new flags from source
+func (f Flags) Add(logger *logger.Logger, source string, ff Flags) map[string]interface{} {
+	notifications := map[string]interface{}{}
+
+	for k, newFlag := range ff.Flags {
+		if storedFlag, ok := f.Flags[k]; ok && storedFlag.Source != source {
+			logger.Warn(fmt.Sprintf(
+				"flag with key %s from source %s already exist, overriding this with flag from source %s",
```

Comment:
do we want to document this behaviour? Or, expand the new signatures to add a priority, e.g. pass the uri argument index to maintain a priority order, preventing lower priority indexes from overwriting existing values? (this may sit outside of the scope of this PR)

```
flagd start --uri file:example_flags.json --uri file:example_flags_lowest_priority.json
```
in this example updates to the `example_flags_lowest_priority.json` source would not be able to overwrite values from `example_flags.json`

Reply:
IMO setting a priority makes the implementation complex (i.e - logic to validate priority and store the priority alongside the configuration source). 

We already highlight something similar for startup configurations [1]

> Using multiple providers is supported however if flag keys are duplicated across multiple sources it may lead to unexpected behavior

Do you think we should improve this description ?

[1] - https://github.com/open-feature/flagd/blob/main/cmd/start.go#L46

Reply:
You're right, no I think what we have here is OK

Reply:
Yep I agree. I think duped flags are enough of a weird edge case that we don't need to define it.